pkg/cache/cache.go

```diff
@@ -327,6 +402,21 @@ func (c *itemCache) removeSecondaryKey(secondaryKey string) error {
 	return nil
 }
 
+//deleteForeignKey - removes a foreign key reference in the cache, but locks the items before doing so
+func (c *itemCache) deleteForeignKey(thisAction cacheAction) (thisReply cacheReply) {
+	key := thisAction.key
+
+	item, ok := c.Items[key]
+	// Check that the key given is in the cache
+	if !ok {
+		thisReply.err = fmt.Errorf("Cache item with key %s does not exist", key)
+		return
+	}
+
+	item.ForeignKey = ""
+	return
```

Comment:
do you have to instantiate thisReply if ok?  Or does it just pass back an "empty" thisReply?

Reply:
Empty thisreply
I just reset the foreign key to be empty